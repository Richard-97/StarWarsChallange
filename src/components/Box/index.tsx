import { View } from 'react-native';
import styled from 'styled-components';
import {
  compose,
  display,
  DisplayProps,
  space,
  SpaceProps,
  position,
  PositionProps,
  flexbox,
  FlexboxProps,
  background,
  BackgroundProps,
  border,
  BorderProps,
  layout,
  LayoutProps,
  BorderRadiusProps,
  boxShadow,
  BoxShadowProps,
  grid,
  GridProps,
} from 'styled-system';

export interface BoxProps
  extends DisplayProps,
    SpaceProps,
    PositionProps,
    FlexboxProps,
    BackgroundProps,
    LayoutProps,
    BorderProps,
    BoxShadowProps,
    GridProps,
    BorderRadiusProps {}

export const Box = styled(View)<BoxProps>`
  ${compose(display, space, position, flexbox, background, border, layout, boxShadow, grid)}
`;
