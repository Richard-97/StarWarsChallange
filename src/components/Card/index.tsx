import React, { FC, ReactNode } from 'react';
import { Box, BoxProps } from 'src/components/Box';
import { BOX_SHADOWS, COLORS, RADIUS, SPACE } from 'src/theme';

interface Props extends BoxProps {
  children: ReactNode;
}

export const Card: FC<Props> = ({ children, ...boxProps }) => (
  <Box
    p={SPACE.M}
    alignItems="center"
    my={SPACE.XS}
    mx={SPACE.S}
    boxShadow={BOX_SHADOWS.CARD}
    borderRadius={RADIUS.RADIUS_10}
    background={COLORS.TERTIARY}
    {...boxProps}
  >
    {children}
  </Box>
);
