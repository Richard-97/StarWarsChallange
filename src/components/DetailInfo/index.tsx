import React, { FC } from 'react';
import { Text } from 'src/components/Text';
import { Box, BoxProps } from 'src/components/Box';
import { COLORS, RADIUS, SPACE } from 'src/theme';

interface Props extends BoxProps {
  title: string;
  label: string;
}

export const DetailInfo: FC<Props> = ({ title, label, ...boxProps }) => (
  <Box
    p={SPACE.S}
    background={COLORS.SECONDARY}
    borderRadius={RADIUS.RADIUS_10}
    alignItems="center"
    justifyContent="center"
    {...boxProps}
  >
    <Text color={COLORS.TERTIARY} mb={SPACE.XS} fontWeight="SEMI_BOLD">
      {title}
    </Text>
    <Text color={COLORS.TERTIARY}>{label}</Text>
  </Box>
);
