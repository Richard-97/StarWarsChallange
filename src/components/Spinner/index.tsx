import React from 'react';
import { View } from 'react-native';
import RNSpinner, { SpinnerProps, SpinnerType } from 'react-native-spinkit';
import { Box } from 'src/components/Box';
import { COLORS } from 'src/theme';

interface Props extends SpinnerProps {
  isFullScreen?: boolean;
  type?: SpinnerType;
}

export const Spinner: React.FC<Props> = ({ isFullScreen, type, ...spinnerProps }) => {
  if (isFullScreen) {
    return (
      <View>
        <Box height="100%" alignItems="center" justifyContent="center">
          <RNSpinner
            type={type ?? '9CubeGrid'}
            color={COLORS.SECONDARY}
            size={50}
            {...spinnerProps}
          />
        </Box>
      </View>
    );
  }
  return <RNSpinner type={type ?? 'ThreeBounce'} color={COLORS.BLACK} {...spinnerProps} />;
};
