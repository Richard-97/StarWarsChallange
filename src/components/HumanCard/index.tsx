import React, { FC } from 'react';
import { SwapiHuman } from 'src/interfaces/swapi';
import { Text } from 'src/components/Text';
import { Card } from 'src/components/Card';
import { Box } from 'src/components/Box';
import { DetailInfo } from 'src/components/DetailInfo';
import { COLORS, SPACE } from 'src/theme';

export const HumanCard: FC<SwapiHuman> = ({ name, height, mass, birth_year, gender }) => (
  <Card>
    <Text fontSize="XXL" fontWeight="BOLD" mb={SPACE.M} color={COLORS.PRIMARY}>
      {name}
    </Text>
    <Box flexDirection="row">
      <Box flex={1} mr={SPACE.XS}>
        <DetailInfo title="Height" label={height} mb={SPACE.S} />
        <DetailInfo title="Weight" label={mass} />
      </Box>
      <Box flex={1} ml={SPACE.XS}>
        <DetailInfo title="Gender" label={gender} mb={SPACE.S} />
        <DetailInfo title="Birth year" label={birth_year} />
      </Box>
    </Box>
  </Card>
);
