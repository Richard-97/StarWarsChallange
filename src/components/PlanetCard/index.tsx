import React, { FC } from 'react';
import { SwapiPlanet } from 'src/interfaces/swapi';
import { Text } from 'src/components/Text';
import { Card } from 'src/components/Card';
import { Box } from 'src/components/Box';
import { DetailInfo } from 'src/components/DetailInfo';
import { COLORS, SPACE } from 'src/theme';

export const PlanetCard: FC<SwapiPlanet> = ({ name, terrain, climate, gravity, population }) => (
  <Card>
    <Text fontSize="XXL" fontWeight="BOLD" mb={SPACE.M} color={COLORS.PRIMARY}>
      {name}
    </Text>
    <Box flexDirection="row">
      <Box flex={1} mr={SPACE.XS}>
        <DetailInfo title="Terrain" label={terrain} mb={SPACE.S} />
        <DetailInfo title="Climate" label={climate} />
      </Box>
      <Box flex={1} ml={SPACE.XS}>
        <DetailInfo title="Gravity" label={gravity} mb={SPACE.S} />
        <DetailInfo title="Population" label={population} />
      </Box>
    </Box>
  </Card>
);
