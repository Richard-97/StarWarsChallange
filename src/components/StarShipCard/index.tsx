import React, { FC } from 'react';
import { SwapiStarShip } from 'src/interfaces/swapi';
import { Card } from 'src/components/Card';
import { Box } from 'src/components/Box';
import { DetailInfo } from 'src/components/DetailInfo';
import { Heading, SubHeading } from 'src/components/Typography';
import { SPACE } from 'src/theme';

export const StarShipCard: FC<SwapiStarShip> = ({
  name,
  manufacturer,
  cost_in_credits,
  length,
  max_atmosphering_speed,
  crew,
  passengers,
  cargo_capacity,
}) => (
  <Card>
    <Heading textAlign="center" mb={SPACE.XS}>
      {name}
    </Heading>
    <SubHeading mb={SPACE.M} textAlign="center">
      {manufacturer}
    </SubHeading>
    <Box width="100%">
      <Box flexDirection="row" mb={SPACE.S}>
        <DetailInfo title="Cost" label={cost_in_credits} flex={1} mr={SPACE.XS} />
        <DetailInfo title="Length" label={length} flex={1} ml={SPACE.XS} />
      </Box>
      <Box flexDirection="row" mb={SPACE.S}>
        <DetailInfo title="Max Speed" label={max_atmosphering_speed} flex={1} mr={SPACE.XS} />
        <DetailInfo title="Crew" label={crew} flex={1} ml={SPACE.XS} />
      </Box>
      <Box flexDirection="row" mb={SPACE.S}>
        <DetailInfo title="Passengers" label={passengers} flex={1} mr={SPACE.XS} />
        <DetailInfo title="Capacity" label={cargo_capacity} flex={1} ml={SPACE.XS} />
      </Box>
    </Box>
  </Card>
);
