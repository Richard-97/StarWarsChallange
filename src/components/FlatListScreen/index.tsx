import React from 'react';
import styled from 'styled-components/native';
import { FlatList, ListRenderItem, RefreshControl, TouchableOpacity } from 'react-native';
import { Box } from 'src/components/Box';
import { Text } from 'src/components/Text';
import { Spinner } from 'src/components/Spinner';
import { FLAT_LIST_END_REACHED_THRESHOLD } from 'src/constants/common';
import { COLORS, SPACE } from 'src/theme';

const BOTTOM_PADDING = 8;

const Container = styled.SafeAreaView`
  height: 100%;
  background-color: ${COLORS.WHITE};
`;

const DataLoadSpinner = () => (
  <Box alignItems="center">
    <Spinner />
  </Box>
);

interface Props<T> {
  isLoading: boolean;
  canFetchMore: boolean;
  data?: T[];
  keyExtractor?: (item: T) => string;
  itemRenderer: ListRenderItem<T>;
  onEndReached: () => Promise<void>;
  refetch: () => void;
  noDataLabel: string;
}

export const FlatListScreen = <T extends Record<string, any>>({
  noDataLabel,
  isLoading,
  canFetchMore,
  data,
  keyExtractor = ({ id }) => id as string,
  itemRenderer,
  onEndReached,
  refetch,
}: Props<T>) => {
  return (
    <Box>
      <Container>
        {isLoading && !data ? (
          <DataLoadSpinner />
        ) : (
          <FlatList
            overScrollMode="never"
            ListEmptyComponent={
              isLoading ? (
                <DataLoadSpinner />
              ) : (
                <>
                  <Box as={TouchableOpacity} alignItems="center" onPress={() => refetch()}>
                    <Text>Refetching...</Text>
                  </Box>
                  <Text mb={SPACE.M} fontSize="M" textAlign="center">
                    {noDataLabel}
                  </Text>
                </>
              )
            }
            data={data}
            keyExtractor={keyExtractor}
            renderItem={itemRenderer}
            showsVerticalScrollIndicator={false}
            onEndReachedThreshold={FLAT_LIST_END_REACHED_THRESHOLD}
            onEndReached={async () => {
              if (canFetchMore) {
                await onEndReached();
              }
            }}
            {...(isLoading && canFetchMore && { ListFooterComponent: <DataLoadSpinner /> })}
            contentContainerStyle={{ paddingBottom: BOTTOM_PADDING }}
            refreshControl={
              <RefreshControl refreshing={isLoading} onRefresh={refetch} tintColor={COLORS.BLACK} />
            }
          />
        )}
      </Container>
    </Box>
  );
};
