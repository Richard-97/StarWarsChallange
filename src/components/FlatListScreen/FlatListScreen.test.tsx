import React from 'react';
import { cleanup, render } from '@testing-library/react-native';
import { FlatListScreen } from 'src/components/FlatListScreen';
import { HumanCard } from 'src/components/HumanCard';

const MOCKED_DATA = [
  { id: 1, name: 'John', mass: '74', height: '178', birth_year: '112BBY', gender: 'male' },
  { id: 2, name: 'Peter', mass: '86', height: '182', birth_year: '111BBY', gender: 'n/a' },
  { id: 3, name: 'Marco', mass: '89', height: '171', birth_year: '114BBY', gender: 'female' },
];

afterEach(() => {
  cleanup();
});

describe('FlatListScreen', () => {
  const mockFn = jest.fn();

  it('matches DOM Snapshot', () => {
    const tree = render(
      <FlatListScreen
        isLoading={true}
        data={MOCKED_DATA}
        noDataLabel="No data"
        canFetchMore={false}
        itemRenderer={({ item: { id, ...props } }) => <HumanCard key={id} {...props} />}
        onEndReached={async () => {
          mockFn();
        }}
        refetch={mockFn}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('render children correctly', () => {
    const { getByText } = render(
      <FlatListScreen
        isLoading={true}
        data={MOCKED_DATA}
        noDataLabel="No data"
        canFetchMore={false}
        itemRenderer={({ item: { id, ...props } }) => <HumanCard key={id} {...props} />}
        onEndReached={async () => {
          mockFn();
        }}
        refetch={mockFn}
      />
    );

    expect(getByText(MOCKED_DATA[0].name)).toBeDefined();
    expect(getByText(MOCKED_DATA[1].name)).toBeDefined();
    expect(getByText(MOCKED_DATA[2].name)).toBeDefined();
  });
});
