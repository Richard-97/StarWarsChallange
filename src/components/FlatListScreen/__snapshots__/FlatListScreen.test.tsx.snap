// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlatListScreen matches DOM Snapshot 1`] = `
<View
  className="sc-bdvvtL bExxtu"
>
  <RCTSafeAreaView
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
          "height": "100%",
        },
      ]
    }
  >
    <RCTScrollView
      ListEmptyComponent={<DataLoadSpinner />}
      contentContainerStyle={
        {
          "paddingBottom": 8,
        }
      }
      data={
        [
          {
            "birth_year": "112BBY",
            "gender": "male",
            "height": "178",
            "id": 1,
            "mass": "74",
            "name": "John",
          },
          {
            "birth_year": "111BBY",
            "gender": "n/a",
            "height": "182",
            "id": 2,
            "mass": "86",
            "name": "Peter",
          },
          {
            "birth_year": "114BBY",
            "gender": "female",
            "height": "171",
            "id": 3,
            "mass": "89",
            "name": "Marco",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onEndReached={[Function]}
      onEndReachedThreshold={0.5}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      overScrollMode="never"
      refreshControl={
        <RefreshControlMock
          onRefresh={[MockFunction]}
          refreshing={true}
          tintColor="#000"
        />
      }
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <RCTRefreshControl />
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            alignItems="center"
            background="#201652"
            borderRadius="10px"
            boxShadow="5px 3px 5px #e5e5e5"
            className="sc-bdvvtL iFUkdj"
            mx="16px"
            my="8px"
            p="24px"
          >
            <Text
              color="#FBE123"
              fontFamily="Roboto"
              fontSize="XXL"
              fontWeight="BOLD"
              mb="24px"
              style={
                [
                  {
                    "color": "#FBE123",
                    "fontFamily": "Roboto",
                    "fontSize": 25,
                    "fontWeight": "700",
                    "marginBottom": 24,
                  },
                ]
              }
            >
              John
            </Text>
            <View
              className="sc-bdvvtL hrMiJy"
              flexDirection="row"
            >
              <View
                className="sc-bdvvtL fzpeaf"
                flex={1}
                mr="8px"
              >
                <View
                  alignItems="center"
                  background="#3DBFAF"
                  borderRadius="10px"
                  className="sc-bdvvtL hVylIW"
                  justifyContent="center"
                  mb="16px"
                  p="16px"
                >
                  <Text
                    color="#201652"
                    fontFamily="Roboto"
                    fontSize="M"
                    fontWeight="SEMI_BOLD"
                    mb="8px"
                    style={
                      [
                        {
                          "color": "#201652",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "600",
                          "marginBottom": 8,
                        },
                      ]
                    }
                  >
                    Height
                  </Text>
                  <Text
                    color="#201652"
                    fontFamily="Roboto"
                    fontSize="M"
                    fontWeight="NORMAL"
                    style={
                      [
                        {
                          "color": "#201652",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "400",
                        },
                      ]
                    }
                  >
                    178
                  </Text>
                </View>
                <View
                  alignItems="center"
                  background="#3DBFAF"
                  borderRadius="10px"
                  className="sc-bdvvtL jfUa-dE"
                  justifyContent="center"
                  p="16px"
                >
                  <Text
                    color="#201652"
                    fontFamily="Roboto"
                    fontSize="M"
                    fontWeight="SEMI_BOLD"
                    mb="8px"
                    style={
                      [
                        {
                          "color": "#201652",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "600",
                          "marginBottom": 8,
                        },
                      ]
                    }
                  >
                    Weight
                  </Text>
                  <Text
                    color="#201652"
                    fontFamily="Roboto"
                    fontSize="M"
                    fontWeight="NORMAL"
                    style={
                      [
                        {
                          "color": "#201652",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "400",
                        },
                      ]
                    }
                  >
                    74
                  </Text>
                </View>
              </View>
              <View
                className="sc-bdvvtL hXqByU"
                flex={1}
                ml="8px"
              >
                <View
                  alignItems="center"
                  background="#3DBFAF"
                  borderRadius="10px"
                  className="sc-bdvvtL hVylIW"
                  justifyContent="center"
                  mb="16px"
                  p="16px"
                >
                  <Text
                    color="#201652"
                    fontFamily="Roboto"
                    fontSize="M"
                    fontWeight="SEMI_BOLD"
                    mb="8px"
                    style={
                      [
                        {
                          "color": "#201652",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "600",
                          "marginBottom": 8,
                        },
                      ]
                    }
                  >
                    Gender
                  </Text>
                  <Text
                    color="#201652"
                    fontFamily="Roboto"
                    fontSize="M"
                    fontWeight="NORMAL"
                    style={
                      [
                        {
                          "color": "#201652",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "400",
                        },
                      ]
                    }
                  >
                    male
                  </Text>
                </View>
                <View
                  alignItems="center"
                  background="#3DBFAF"
                  borderRadius="10px"
                  className="sc-bdvvtL jfUa-dE"
                  justifyContent="center"
                  p="16px"
                >
                  <Text
                    color="#201652"
                    fontFamily="Roboto"
                    fontSize="M"
                    fontWeight="SEMI_BOLD"
                    mb="8px"
                    style={
                      [
                        {
                          "color": "#201652",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "600",
                          "marginBottom": 8,
                        },
                      ]
                    }
                  >
                    Birth year
                  </Text>
                  <Text
                    color="#201652"
                    fontFamily="Roboto"
                    fontSize="M"
                    fontWeight="NORMAL"
                    style={
                      [
                        {
                          "color": "#201652",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "400",
                        },
                      ]
                    }
                  >
                    112BBY
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            alignItems="center"
            background="#201652"
            borderRadius="10px"
            boxShadow="5px 3px 5px #e5e5e5"
            className="sc-bdvvtL iFUkdj"
            mx="16px"
            my="8px"
            p="24px"
          >
            <Text
              color="#FBE123"
              fontFamily="Roboto"
              fontSize="XXL"
              fontWeight="BOLD"
              mb="24px"
              style={
                [
                  {
                    "color": "#FBE123",
                    "fontFamily": "Roboto",
                    "fontSize": 25,
                    "fontWeight": "700",
                    "marginBottom": 24,
                  },
                ]
              }
            >
              Peter
            </Text>
            <View
              className="sc-bdvvtL hrMiJy"
              flexDirection="row"
            >
              <View
                className="sc-bdvvtL fzpeaf"
                flex={1}
                mr="8px"
              >
                <View
                  alignItems="center"
                  background="#3DBFAF"
                  borderRadius="10px"
                  className="sc-bdvvtL hVylIW"
                  justifyContent="center"
                  mb="16px"
                  p="16px"
                >
                  <Text
                    color="#201652"
                    fontFamily="Roboto"
                    fontSize="M"
                    fontWeight="SEMI_BOLD"
                    mb="8px"
                    style={
                      [
                        {
                          "color": "#201652",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "600",
                          "marginBottom": 8,
                        },
                      ]
                    }
                  >
                    Height
                  </Text>
                  <Text
                    color="#201652"
                    fontFamily="Roboto"
                    fontSize="M"
                    fontWeight="NORMAL"
                    style={
                      [
                        {
                          "color": "#201652",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "400",
                        },
                      ]
                    }
                  >
                    182
                  </Text>
                </View>
                <View
                  alignItems="center"
                  background="#3DBFAF"
                  borderRadius="10px"
                  className="sc-bdvvtL jfUa-dE"
                  justifyContent="center"
                  p="16px"
                >
                  <Text
                    color="#201652"
                    fontFamily="Roboto"
                    fontSize="M"
                    fontWeight="SEMI_BOLD"
                    mb="8px"
                    style={
                      [
                        {
                          "color": "#201652",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "600",
                          "marginBottom": 8,
                        },
                      ]
                    }
                  >
                    Weight
                  </Text>
                  <Text
                    color="#201652"
                    fontFamily="Roboto"
                    fontSize="M"
                    fontWeight="NORMAL"
                    style={
                      [
                        {
                          "color": "#201652",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "400",
                        },
                      ]
                    }
                  >
                    86
                  </Text>
                </View>
              </View>
              <View
                className="sc-bdvvtL hXqByU"
                flex={1}
                ml="8px"
              >
                <View
                  alignItems="center"
                  background="#3DBFAF"
                  borderRadius="10px"
                  className="sc-bdvvtL hVylIW"
                  justifyContent="center"
                  mb="16px"
                  p="16px"
                >
                  <Text
                    color="#201652"
                    fontFamily="Roboto"
                    fontSize="M"
                    fontWeight="SEMI_BOLD"
                    mb="8px"
                    style={
                      [
                        {
                          "color": "#201652",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "600",
                          "marginBottom": 8,
                        },
                      ]
                    }
                  >
                    Gender
                  </Text>
                  <Text
                    color="#201652"
                    fontFamily="Roboto"
                    fontSize="M"
                    fontWeight="NORMAL"
                    style={
                      [
                        {
                          "color": "#201652",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "400",
                        },
                      ]
                    }
                  >
                    n/a
                  </Text>
                </View>
                <View
                  alignItems="center"
                  background="#3DBFAF"
                  borderRadius="10px"
                  className="sc-bdvvtL jfUa-dE"
                  justifyContent="center"
                  p="16px"
                >
                  <Text
                    color="#201652"
                    fontFamily="Roboto"
                    fontSize="M"
                    fontWeight="SEMI_BOLD"
                    mb="8px"
                    style={
                      [
                        {
                          "color": "#201652",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "600",
                          "marginBottom": 8,
                        },
                      ]
                    }
                  >
                    Birth year
                  </Text>
                  <Text
                    color="#201652"
                    fontFamily="Roboto"
                    fontSize="M"
                    fontWeight="NORMAL"
                    style={
                      [
                        {
                          "color": "#201652",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "400",
                        },
                      ]
                    }
                  >
                    111BBY
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            alignItems="center"
            background="#201652"
            borderRadius="10px"
            boxShadow="5px 3px 5px #e5e5e5"
            className="sc-bdvvtL iFUkdj"
            mx="16px"
            my="8px"
            p="24px"
          >
            <Text
              color="#FBE123"
              fontFamily="Roboto"
              fontSize="XXL"
              fontWeight="BOLD"
              mb="24px"
              style={
                [
                  {
                    "color": "#FBE123",
                    "fontFamily": "Roboto",
                    "fontSize": 25,
                    "fontWeight": "700",
                    "marginBottom": 24,
                  },
                ]
              }
            >
              Marco
            </Text>
            <View
              className="sc-bdvvtL hrMiJy"
              flexDirection="row"
            >
              <View
                className="sc-bdvvtL fzpeaf"
                flex={1}
                mr="8px"
              >
                <View
                  alignItems="center"
                  background="#3DBFAF"
                  borderRadius="10px"
                  className="sc-bdvvtL hVylIW"
                  justifyContent="center"
                  mb="16px"
                  p="16px"
                >
                  <Text
                    color="#201652"
                    fontFamily="Roboto"
                    fontSize="M"
                    fontWeight="SEMI_BOLD"
                    mb="8px"
                    style={
                      [
                        {
                          "color": "#201652",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "600",
                          "marginBottom": 8,
                        },
                      ]
                    }
                  >
                    Height
                  </Text>
                  <Text
                    color="#201652"
                    fontFamily="Roboto"
                    fontSize="M"
                    fontWeight="NORMAL"
                    style={
                      [
                        {
                          "color": "#201652",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "400",
                        },
                      ]
                    }
                  >
                    171
                  </Text>
                </View>
                <View
                  alignItems="center"
                  background="#3DBFAF"
                  borderRadius="10px"
                  className="sc-bdvvtL jfUa-dE"
                  justifyContent="center"
                  p="16px"
                >
                  <Text
                    color="#201652"
                    fontFamily="Roboto"
                    fontSize="M"
                    fontWeight="SEMI_BOLD"
                    mb="8px"
                    style={
                      [
                        {
                          "color": "#201652",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "600",
                          "marginBottom": 8,
                        },
                      ]
                    }
                  >
                    Weight
                  </Text>
                  <Text
                    color="#201652"
                    fontFamily="Roboto"
                    fontSize="M"
                    fontWeight="NORMAL"
                    style={
                      [
                        {
                          "color": "#201652",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "400",
                        },
                      ]
                    }
                  >
                    89
                  </Text>
                </View>
              </View>
              <View
                className="sc-bdvvtL hXqByU"
                flex={1}
                ml="8px"
              >
                <View
                  alignItems="center"
                  background="#3DBFAF"
                  borderRadius="10px"
                  className="sc-bdvvtL hVylIW"
                  justifyContent="center"
                  mb="16px"
                  p="16px"
                >
                  <Text
                    color="#201652"
                    fontFamily="Roboto"
                    fontSize="M"
                    fontWeight="SEMI_BOLD"
                    mb="8px"
                    style={
                      [
                        {
                          "color": "#201652",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "600",
                          "marginBottom": 8,
                        },
                      ]
                    }
                  >
                    Gender
                  </Text>
                  <Text
                    color="#201652"
                    fontFamily="Roboto"
                    fontSize="M"
                    fontWeight="NORMAL"
                    style={
                      [
                        {
                          "color": "#201652",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "400",
                        },
                      ]
                    }
                  >
                    female
                  </Text>
                </View>
                <View
                  alignItems="center"
                  background="#3DBFAF"
                  borderRadius="10px"
                  className="sc-bdvvtL jfUa-dE"
                  justifyContent="center"
                  p="16px"
                >
                  <Text
                    color="#201652"
                    fontFamily="Roboto"
                    fontSize="M"
                    fontWeight="SEMI_BOLD"
                    mb="8px"
                    style={
                      [
                        {
                          "color": "#201652",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "600",
                          "marginBottom": 8,
                        },
                      ]
                    }
                  >
                    Birth year
                  </Text>
                  <Text
                    color="#201652"
                    fontFamily="Roboto"
                    fontSize="M"
                    fontWeight="NORMAL"
                    style={
                      [
                        {
                          "color": "#201652",
                          "fontFamily": "Roboto",
                          "fontSize": 16,
                          "fontWeight": "400",
                        },
                      ]
                    }
                  >
                    114BBY
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
</View>
`;
